### DISCUSSION TITLE
Pytest junitxml plugin error vs failure question

### DISCUSSION THREAD

**Discussion by niroshaimos** [pytest-dev/pytest#7950] [Q&A] [answered]
Hi! Iv'e been messing around with the junitxml plugin for pytest and after reading a bit in the junitxml documentation I understand that the difference between error and failure is if the test failed on an assertion or on a random error.
However, whenever a test of mine fails (on an assertion error or another random error) in the junitxml file the result is always the same. The test is considered a failure and never an error. 
My question is: how do I differentiate between a test failed because of an assertion error and a test failed because of a problem in the test itself? Did i miss something about the junitxml error portion?
Thanks in advance!

**Comment by nicoddemus** [comment, marked as answer]
Hi @niroshaimos,

pytest considers a _failure_ any assertion error or exception raised inside a test function; _errors_ happen when an assertion error or exception is raised during setup/teardown/collection.

When writing the junitxml file, we follow the same semantics, writing an `error` or `failure` tag accordingly.

**Reply by niroshaimos** [reply]
Thanks!
